### ISSUE TITLE
Unexpected structure and dominance of unknown latent at scale (700k cells) with vanilla Biolord settings

### ISSUE THREAD

**Issue by regisscar** [nitzanlab/biolord#23]
### Report

I am observing a qualitative change in the behavior of the unknown latent and the concatenated latent when scaling Biolord from a medium-sized dataset (~60k cells) to a large dataset (~700k cells), using normalized input data and completely vanilla Biolord settings.

For the smaller dataset, the unknown latent shows the expected roughly isotropic Gaussian-like blob. For the larger dataset, the unknown latent instead shows a Gaussian core plus additional apparent structure. When coloring this latent space by all available biological and technical labels (e.g. condition, sample, cell type, counts, mitochondrial genes, etc..), this additional structure does not align with any of them.

At the same time, the concatenated latent for the large dataset visually resembles the unknown latent much more than the known sublatents (expected due to the larger size of the the uknown latent) . The known sublatents do show clean and separation when visualized on their own, but appear to contribute little to the geometry of the concatenated latent. This whole behavior remains the same regardless of how many covariates are included in the model.

Training and validation are stable, and all individual loss components were monitored directly. There are no signs of divergence or instability. This suggests that the behavior is likely not due to a training failure, but rather related to how the per-cell optimized unknown latent behaves when scaling to very large datasets. At this scale, the unknown latent may have enough capacity to capture weak residual variation (biological or technical), which then strongly influences the geometry of the concatenated latent without clearly aligning to the modeled covariates.

I was wondering whether this is expected behavior when applying Biolord to hundreds of thousands of cells, and if there are recommended settings or best practices to keep the unknown latent from dominating the concatenated representation at this scale.

Figures showing the unknown latent, known sublatents, and concatenated latent for both dataset sizes are attached.

Uknown latent:

<img width="441" height="662" alt="Image" src="https://github.com/user-attachments/assets/05e87001-8c31-4879-90a1-8b7424916701" />

Example of individual covariate latent block:

<img width="339" height="519" alt="Image" src="https://github.com/user-attachments/assets/a1145bfe-9fbd-436b-9e26-80460b1a0e25" />

Full concatanated latent

<img width="343" height="518" alt="Image" src="https://github.com/user-attachments/assets/87d687dc-6013-48e5-b82b-efa7e5c8a2e9" />

Training losses where the steps = epochs

<img width="1153" height="671" alt="Image" src="https://github.com/user-attachments/assets/a8a60479-e81b-4b5c-afce-9532ab60d491" />

Validation losses with steps = epochs 

<img width="1160" height="685" alt="Image" src="https://github.com/user-attachments/assets/e2b81337-3aba-4fdf-aab2-44c7735fb608" />

**Comment by zoepiran** [comment]
Dear regisscar, 

Thank you for your interest in biolord and detailed exploration. I am still afraid that without more concrete biological context we cant provide a complete response. With that I am afraid i do not follow exactly what is the core "wrong" observation you are reporting. 

I believe that though depicting slightly more structure It is positive that the unknown latents are not aligned  with known biological factors and capture indeed unknown variation. With that given your intuition which seems accurate that capacity constraints are controlling the behavior i would advice modifying the hyperparameters (e.g. `unknown_attribute_noise_param` and `unknown_attribute_penalty`)

Please feel free to follow up with additional questions.
